### PR TITLE
Date junction

### PR DESCRIPTION
**Pull request by laythq**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -34,6 +35,7 @@ class MakersBNB < Sinatra::Base
     @user_listings = Listing.user_listings(session[:id])
     requests = Request.my_requests(@user.id)
     @my_requests = requests.map {|request| Listing.retrieve_listing(request.listing_id)}
+    p @available_dates = @user_listings.map { |listing| ListingsDates.return_dates(listing)}
```
Comment (elishkaflint) [inline app.rb]: take out the p

File: app.rb [hunk 2]
```diff
@@ -88,7 +90,12 @@ class MakersBNB < Sinatra::Base
   end
 
   post '/add_listing' do
+    p params[:date]
```
Comment (elishkaflint) [inline app.rb]: take out the p

File: app.rb [hunk 3]
```diff
@@ -88,7 +90,12 @@ class MakersBNB < Sinatra::Base
...
     listing = Listing.create(params[:name], session[:id], params[:description], params[:price])
+    date = Dates.create(params[:date])
+    date.id
+    p ListingsDates.add_date((date.id).to_i, listing.id)
```
Comment (elishkaflint) [inline app.rb]: take out the p

File: app.rb [hunk 4]
```diff
@@ -88,7 +90,12 @@ class MakersBNB < Sinatra::Base
   end
 
   post '/add_listing' do
+    p params[:date]
     listing = Listing.create(params[:name], session[:id], params[:description], params[:price])
+    date = Dates.create(params[:date])
+    date.id
```
Comment (elishkaflint) [inline app.rb]: don't think we need this line?

File: spec/features/addDateSpec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+feature 'Add a Date' do
+  scenario 'User adds available date to his existing listing' do
+    visit('/')
```
Comment (elishkaflint) [inline spec/features/addDateSpec.rb]: should refactor this into a spec helper

Comment (laythq) [reply spec/features/addDateSpec.rb]: will do

**Pull request merged**